packages/muelu/src/MueCentral/MueLu_VerbosityLevel.hpp

```diff
@@ -109,6 +110,40 @@ namespace MueLu {
   //!
   VerbLevel toMueLuVerbLevel(const Teuchos::EVerbosityLevel verbLevel);
 
+  //! global variable that holds the default output rank for new VerboseObjects
+  extern int defaultMueLuOutputRank;
+
+  //! constant that is interpreted as "output on default rank"
+  extern const int OUTPUT_TO_ROOT;
+
+  //! On all MPI ranks that are contained in oldMap, set the default output rank for MueLu to the lowest available rank that is present in newMap.
+  //! This is used in rebalancing, since the original root output might have been dropped.
+  template <class LocalOrdinal, class GlobalOrdinal, class Node>
+  void updateDefaultMueLuOutputRank(Teuchos::RCP<const typename Xpetra::Map<LocalOrdinal, GlobalOrdinal, Node> > oldMap,
+                                    Teuchos::RCP<const typename Xpetra::Map<LocalOrdinal, GlobalOrdinal, Node> > newMap) {
+#ifdef HAVE_MPI
+    Teuchos::RCP<Teuchos::Comm<int> > oldComm = Teuchos::rcp_const_cast<Teuchos::Comm<int> >(oldMap->getComm());
+    if (newMap == Teuchos::null || oldComm->getSize() != newMap->getComm()->getSize()) {
+      int mpiStarted = 0;
+      MPI_Initialized(&mpiStarted);
```

Comment:
If MPI exists but has not been initialized, then the comms _must_ be SerialComm.  In that case, both comms would have size 1, so you could never reach this code anyway.